edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/storage/MessageStore.cs

```diff
@@ -523,6 +513,37 @@ public MessageRef(string edgeMessageId, DateTime timeStamp)
             public DateTime TimeStamp { get; }
         }
 
+        class MessageWithOffset : IMessage
```

Comment:
Is the point of this to allow you to change the offset of a readonly message? If so, please add a small comment. MessageWithOffset isn't very descriptive since all messages have an offset.